Area51/Sources/SubredditsViewController.swift

```diff
@@ -16,28 +25,63 @@ class SubredditsViewController: UIViewController {
     private func setupTableView() {
         self.tableView.delegate = self
         self.tableView.dataSource = self
+        self.tableView.refreshControl = UIRefreshControl()
+        self.tableView.refreshControl?.addTarget(self, action: #selector(self.refreshData), for: .valueChanged)
     }
 
     private func showSubreddit(_ subreddit: Subreddit) {
         let feedViewController = FeedViewController.instantiateFromStoryboard(withSubreddit: subreddit)
         self.splitViewController?.showDetailViewController(feedViewController.embeddedInNavigationController(),
                                                            sender: self)
     }
+    @objc
+    private func refreshData() {
+        self.dataSource.refresh()
+    }
 }
 
 extension SubredditsViewController: UITableViewDataSource, UITableViewDelegate {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
+        case 0:
+            return "Reddit Feeds"
+        default:
+            return "Default Subreddits"
+        }
+    }
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return self.subreddits.count
+        switch section {
+        case 0:
+            return self.subreddits.count
+        default:
+            return self.dataSource.listings.count
+        }
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        self.dataSource.loadMoreIfNeeded(currentIndex: indexPath.row)
```

Comment:
Probably do not need this? How many subreddits are coming back?

Reply:
If the number that comes back is greater than single page, I can add a variable to the service which can be set to pull more listings at a single time to handle this case. We should get them all back in a single request.

Reply:
Total count of subreddits is 47. And the number that comes back is greater than a single page

Reply:
@dulithadabare What is the definition of a "single page"?

Reply:
The first time I load the request, I only get back 25 results. The next 22 results is loaded afterwards.
The single page is the number of elements in response you get at one time. Here, it's 25.
Btw, this is how I understood it. I'm not sure and maybe @kgellci can clarify

Reply:
Okay, I understand. 25 is a reasonable amount of elements when it comes to pagination. 

@kgellci - Personally, I feel this is appropriate. If the number of elements did grow in future, users on slower networks would wait longer for a single, larger response, with more elements to come down than a smaller one with less elements for the initial load. 

Ideally, you don't want to keep users waiting too long for data to download - it's more of an issue on slower connections than fast ones.